### PR TITLE
Issue 12215 remove password reset when not possible

### LINKED ISSUE #12215: enable user backends to show a different page when login failed (e.g. password wrong or needs to change)
ownCloud seems not to support password changes for AD users!
If a user is forced by AD to change his password (i.e. after a reset caused by IT-helpdesk), he got to find an alternative way to change his password before he will able to log in to ownCloud again.

Further, ownCloud tries to send a password reset mail in the case above - even to Active Directoy imported users, which makes no sense.

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #12215 

Backport request as per issue @karlitschek 

Review please @blizzz @MorrisJobke @LukasReschke

### CODE REVIEW

File: lib/private/util.php [hunk 1]
```diff
@@ -947,6 +947,14 @@ public static function displayLoginPage($errors = array(), $messages = []) {
 			$parameters['redirect_url'] = $_REQUEST['redirect_url'];
 		}
 
+		$parameters['canResetPassword'] = true;
+		if (!\OC::$server->getSystemConfig()->getValue('lost_password_link')) {
+			$user = \OC::$server->getUserManager()->get($_REQUEST['user']);
```
Comment (blizzz) [inline lib/private/util.php]: won't work reliably. Default LDAP setups will not work with this.

1. the login name input can be case insensitive (to play with first capital letter of mobile devices), opposite to user names.

2. the login name is not necessarily the user name (includes risk of false positives, given the impact is against convenience, but not critical).

3.  I do not have a better idea for this either

Comment (blizzz) [inline lib/private/util.php]: Well, together  with the configurable string it's probably the  best solution we can have.

**Comment by blizzz** [comment]
Works as advertised :+1:

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by MorrisJobke** [comment]
> Backport request as per issue @karlitschek

Ping ;)

**Comment by karlitschek** [comment]
good one. Please backport :+1:

**Pull request merged**